lib/src/reduction_tree.dart

```diff
@@ -54,91 +24,130 @@ class ReductionTree extends Module {
   /// The flop depth of the tree from the output to the leaves.
   int get latency => _computed.flopDepth;
 
+  /// Operation to be performed at each node. Note that [_operation] can widen
+  /// the output. The logic function must support the operation for 2 and up to
+  /// [_radix] inputs.
+  final Logic Function(List<Logic> inputs, {String name}) _operation;
+
+  /// Specified width of input to each reduction node (e.g., binary: radix=2)
+  late final int _radix;
```

Comment:
primitive types like `int` and `bool`, or really any non-hardware-signal/module thing (even `Function`s like `operation`), especially when they are directly passed from the user, can be just public `final` variables with `this.radix` type of syntax in the constructor.  That's also better for when you reference those variables in documentation comments `[radix]` vs `[_radix]`